File: Heath/app/src/main/java/com/example/ngoctta999123/heath/Retrofit2/APIUtils.java
package com.example.ngoctta999123.heath.Retrofit2;

import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

public class APIUtils {
    public static final String Base_Url = "https://fathomless-savannah-38522.herokuapp.com/api/";

    //nhan va gui du lieu di
    public static DataClient getData(Context context){
        SharedPreferences sharedPreferences = context.getSharedPreferences("TOKEN", Context.MODE_PRIVATE);
        String uid = sharedPreferences.getString("ACCESS_TOKEN", "uid");
        Log.d("token",uid);
        return RetrofitClient.getClient(Base_Url, uid).create(DataClient.class);
    }
}

File: MyApplication2/app/src/main/java/com/example/myapplication/ApiUtils.kt
package com.example.myapplication

import com.example.retrofit.WebService

object ApiUtils {
    val BASE_URL = "https://fathomless-savannah-38522.herokuapp.com/api/"
    val webService: WebService = RetrofitClient.getClient(BASE_URL).create<WebService>(WebService::class.java!!)
}